### DISCUSSION TITLE
autoincrement error

### DISCUSSION THREAD

**Discussion by IRediTOTO** [prisma/prisma#14785] [Q&A]
### Bug description

Hi, I am trying create a row with `  id Int @id @default(autoincrement())` 
But I get this error 
```
Invalid `prisma.price.create()` invocation:
Unique constraint failed on the fields: (`id`)
```
But if I run create with custom `id` it work fine 
```
  const createPost = await prm.price.create({
    data: {
      id: 26,  // Remove this will error.
      price: 222,
      codeNum: 3,
      content: "This is basic plan, see image bellow",
    },
  });
```
And after run with custom id, I removed id from data. It still error.

### How to reproduce

I think don't need it, just wonder how to fix it

### Prisma information
```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgresql"
  // url      = env("SUPABASE_DB_URL")
  url      = env("DATABASE_URL")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextSearch"]
}

// --------------------------------------
model VerificationToken {
  identifier String
  token      String   @unique
  expires    DateTime

  @@unique([identifier, token])
}

model Account {
  id                Int     @id @default(autoincrement())
  userId            Int
  type              String
  provider          String
  providerAccountId String
  refresh_token     String? @db.Text
  access_token      String? @db.Text
  expires_at        Int?
  token_type        String?
  scope             String?
  id_token          String? @db.Text
  session_state     String?
  user              User    @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@unique([provider, providerAccountId])
}

model Session {
  id           Int      @id @default(autoincrement())
  sessionToken String   @unique
  userId       Int
  expires      DateTime
  user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)
}

model User {
  id             Int         @id @default(autoincrement())
  createdAt      DateTime    @default(now())
  updatedAt      DateTime    @updatedAt
  name           String      @default("")
  email          String      @unique
  emailVerified  DateTime?
  isVerified     Boolean     @default(false)
  image          String      @default("ltdcollector.com/smile")
  Image          Image       @relation("ImageOnUser", fields: [image], references: [publicId], onDelete: Cascade)
  pdImage        Int?
  PdImage        Image?      @relation("PdImageUser", fields: [pdImage], references: [id], onDelete: Cascade)
  role           UserRole    @default(USER)
  blockRole      BlockRole[] @default([])
  taco           Int         @default(0)
  slot           Int         @default(0)
  medal          Medal[]
  paymentMethod  String[]
  facebook       String?
  showFacebook   Boolean     @default(false)
  paypal         String      @default("")
  showPaypal     Boolean     @default(false)
  publicEmail    String      @default("")
  showEmail      Boolean     @default(false)
  twitter        String      @default("")
  showTwitter    Boolean     @default(false)
  ip             String[]
  description    String      @default("")
  subscribeEmail Boolean?    @default(false)
  //--------------------------------------
  Sessions       Session[]
  Deal           Deal[]
  Accounts       Account[]
  Auction        Auction[]
  CommentAuthor  Comment[]   @relation("CommentAuthor")
  CommentUser    Comment[]   @relation("CommentUser")
  UserVote       UserVote[]  @relation("UserVote")
  AuthorVote     UserVote[]  @relation("AuthorVote")
  Post           Post[]
}

model UserVote {
  id       Int    @id @default(autoincrement())
  review   String @default("")
  rate     Int
  authorId Int?
  Author   User?  @relation("AuthorVote", fields: [authorId], references: [id])
  userId   Int?
  User     User?  @relation("UserVote", fields: [userId], references: [id])
  dealId   Int?
  Deal     Deal?  @relation(fields: [dealId], references: [id])
}

model Deal {
  id          Int        @id @default(autoincrement())
  createdAt   DateTime   @default(now())
  updatedAt   DateTime   @updatedAt
  path        String     @unique
  name        String
  marketUrl   String?    @unique
  dealUrl     String     @unique
  description String     @default("")
  excerpt     String     @default("")
  dealStatus  DealStatus @default(SELLING)
  isActive    Boolean    @default(true)
  isBlocked   Boolean    @default(false)
  planType    PlanType   @default(LIFETIME)
  terms       String[]
  features    String[]
  tldr        String[]

  //-----------------------------------------------------
  authorId Int
  Author   User @relation(fields: [authorId], references: [id])

  //-----------------------------------------------------
  Image Image @relation("ImageOnDeal", fields: [image], references: [id])
  image Int

  //-----------------------------------------------------
  Gallery    Image[]    @relation("GalleryOnDeal")
  // gallery String[]
  //-----------------------------------------------------
  Categories Category[]
  //-----------------------------------------------------

  //-----------------------------------------------------
  Comment  Comment[]
  UserVote UserVote[]

  //-----------------------------------------------------
  DealPrice DealPrice[]
  Post      Post[]
}

model Comment {
  id        Int           @id @default(autoincrement())
  createdAt DateTime      @default(now())
  updatedAt DateTime      @updatedAt
  content   String        @default("")
  status    CommentStatus @default(ACTIVE)
  parentId  Int?
  parent    Comment?      @relation("CommentParentReplies", fields: [parentId], references: [id], onDelete: Cascade)
  replies   Comment[]     @relation("CommentParentReplies")
  like      Int           @default(0)
  unlike    Int           @default(0)
  //-----------------------------------------------------
  authorId  Int
  Author    User          @relation("CommentAuthor", fields: [authorId], references: [id])
  //-----------------------------------------------------
  userId    Int?
  User      User?         @relation("CommentUser", fields: [userId], references: [id])
  //-----------------------------------------------------
  Deal      Deal?         @relation(fields: [dealId], references: [id])
  dealId    Int?
  Post      Post?         @relation(fields: [postId], references: [id])
  postId    Int?
}

model Image {
  id          Int    @id @default(autoincrement())
  publicId    String @unique
  DealImage   Deal[] @relation("ImageOnDeal")
  DealGallery Deal[] @relation("GalleryOnDeal")
  UserImage   User[] @relation("ImageOnUser")
  UserPdImage User[] @relation("PdImageUser")
  Post        Post[] @relation("ImageOnPost")
}

model Category {
  id   Int    @id @default(autoincrement())
  name String @unique
  Deal Deal[]
}

model DealPrice {
  id        Int           @id @default(autoincrement())
  price     Float         @default(0)
  salePrice Float         @default(0)
  features  String[]
  codeNum   Int           @default(1)
  type      DealPriceType @default(CODE)
  content   String        @default("")
  Deal      Deal?         @relation(fields: [dealId], references: [id])
  dealId    Int?
}

enum DealPriceType {
  TIER
  CODE
}

model Price {
  id      Int       @id @default(autoincrement())
  price   Float     @default(0)
  codeNum Int       @default(1)
  content String    @default("")
  Auction Auction[]
  Post    Post?     @relation(fields: [postId], references: [id])
  postId  Int?
}

model Auction {
  id        Int           @id @default(autoincrement())
  priceAuc  Float
  authorId  Int
  priceId   Int
  status    AuctionStatus @default(ACTIVE)
  createdAt DateTime      @default(now())
  updatedAt DateTime      @updatedAt
  Author    User          @relation(fields: [authorId], references: [id])
  Post      Post          @relation(fields: [postId], references: [id])
  postId    Int
  Price     Price         @relation(fields: [priceId], references: [id])
}

model Post {
  id        Int        @id @default(autoincrement())
  title     String     @default("")
  path      String     @unique
  content   String     @default("")
  isAuction Boolean    @default(false)
  status    PostStatus @default(PUBLIC) // For sell or buy
  type      PostType   @default(SELL)
  isBlocked Boolean    @default(false) // Just admin can do

  // -----------------------------------------------------
  Author   User   @relation(fields: [authorId], references: [id])
  authorId Int
  // -----------------------------------------------------
  //-----------------------------------------------------
  Image    Image? @relation("ImageOnPost", fields: [image], references: [id])
  image    Int?
  //-----------------------------------------------------
  dealId   Int
  Deal     Deal   @relation(fields: [dealId], references: [id])

  //-----------------------------------------------------

  Note      Note?     @relation(fields: [noteId], references: [id])
  noteId    Int?      @unique
  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  Comment   Comment[]
  Auction   Auction[]
  Price     Price[]
}

model Note {
  id        Int      @id @default(autoincrement())
  updatedAt DateTime @updatedAt
  content   String?
  Post      Post?
}

enum Status {
  ACTIVE
  INACTIVE
}

enum CommentStatus {
  PENDING
  ACTIVE
  INACTIVE
  SPAM
}

enum Medal {
  OWN_DEAL_10
  OWN_DEAL_20
  OWN_DEAL_30
  OWN_DEAL_40
  OWN_DEAL_50
  STAR_4
  STAR_5
  GET_VOTE_5
  GET_VOTE_10
  GET_VOTE_15
  GET_VOTE_20
  R_VIEW_5
  R_VIEW_10
  R_VIEW_15
  R_VIEW_20
  VERIFIED
  SCAMMER
  ADMIN
}

enum TokenType {
  RESET_PASSWORD
}

enum PaymentMethod {
  PAYPAL
  STRIPE
}

enum UserRole {
  OWNER
  ADMIN
  DEAL_AUTHOR // Can create deal
  USER // Can create Post
  ANONYMOUS
}

enum DealStatus {
  SELLING
  EXPIRED
  EXPIRED_SOON
  COMMING_SOON
  SCAM_WARNING
  SCAM
  DELETE
}

enum PostStatus {
  PUBLIC
  PRIVATE
  AUCTION
  DELETE
}

enum AuctionStatus {
  ACTIVE
  INACTIVE
  SOLD
  CLOSED
  DEALING
}

enum PostType {
  SELL
  BUY
}

enum UserType {
  USER
  DEAL_AUTHOR
  ADMIN
}

enum AccountProvider {
  facebook
  google
  twitter
  vk
  instagram
  yandex
  odnoklassniki
  other
  password
}

enum BlockRole {
  COMMENT
  CREATE_DEAL
  CREATE_POST_DEAL
}

enum PlanType {
  FREE
  LIFETIME
}


```
### Environment & setup

- OS: Supabase
- Database: Supabase
- Node.js version:16

### Prisma Version

```
4.2.1
```

**Comment by janpio** [comment]
Did you migrate your database after definining that the `@id` field should be an auto increment?

**Reply by IRediTOTO** [reply]
Yes, I think. I changed the model name, so I make copy / delete / re-create table... yeah, I copied hole the `id` field. Is there anyway to fix it? Can I create new model, copy table name `abc`, delete table `Price`  then re-create the `Price` table again?

**Reply by janpio** [reply]
Did you run a migration via `prisma migrate deploy`/`prisma migrate dev` or `prisma db push`? Or did you just change the `schema.prisma` file?

**Reply by IRediTOTO** [reply]
I changed prisma file, then run `prisma db push` just that.

**Reply by janpio** [reply]
Yes, that should be enough. That should automatically run `prisma generate` as well - and that you _can_ run a Prisma Client query without having to supply the ID indicates that worked.

I suggest you log in to your database with another database UI (e.g. DBeaver) and check if `Price` is configured correctly to have an auto incrementing ID column.

**Reply by renolation** [reply]
i already have auto increment on db side with nextval('echo_entity_id_seq'::regclass)
but i still have same error

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
